### PR TITLE
Feature/solved problem component

### PR DESCRIPTION
**Pull request by bodichka11**
https://trello.com/c/T6zDAumk/40-profile-solved-problem-component

### CODE REVIEW

File: frontend/src/app/modules/landing/landing.module.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+import { UserModule } from '@modules/user/user.module';
 import { SharedModule } from '@shared/shared.module';
 
 import { LandingPageComponent } from './landing-page/landing-page.component';
 import { LandingRoutingModule } from './landing-routing.module';
 
 @NgModule({
     declarations: [LandingPageComponent],
-    imports: [SharedModule, LandingRoutingModule],
+    imports: [SharedModule, LandingRoutingModule, UserModule],
```
Comment (tatianahutii) [inline frontend/src/app/modules/landing/landing.module.ts:10]: why do we need UserProfile in Landing module?

File: frontend/src/app/shared/components/pie-chart/pie-chart.component.sass [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  align-items: center
+  justify-content: center
+  
+
+  .chart-container
+    position: relative
+    display: inline-block
+    
+    
+
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/pie-chart/pie-chart.component.sass:14]: remove unnecessary empty lines (everywhere)

File: frontend/src/app/shared/components/pie-chart/pie-chart.component.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+@Component({
+    selector: 'app-pie-chart',
+    templateUrl: './pie-chart.component.html',
+    styleUrls: ['./pie-chart.component.sass'],
+})
+export class PieChartComponent implements OnInit {
+    @Input() CurrentNumber: number;
+
+    @Input() TotalNumber: number;
+
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/pie-chart/pie-chart.component.ts:15]:
```suggestion
    @Input() currentNumber: number;

    @Input() totalNumber: number;

```

File: frontend/src/app/shared/components/pie-chart/pie-chart.component.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    @Input() activeColor: string;
+
+    single: PieChartData[] = [];
+
+    colorScheme!: Color;
+
+    view: [number, number] = [140, 140];
+
+    constructor() {
+        Object.assign(this, { this: this.single });
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/pie-chart/pie-chart.component.ts:25]: why do we need this?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/user/solved-problem/solved-problem.component.sass [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  width: 214px
+  height: 36px
+  
+  &:not(:first-child)
+    height: 52px
+    padding-top: 16px
+
+
+::ng-deep .level-card:nth-child(1) .progress-bar
+    background-color: #2CBB5D40
```
Comment (tatianahutii) [inline frontend/src/app/modules/user/solved-problem/solved-problem.component.sass:54]: still a lot of colors need to be replaces with variables

File: frontend/src/app/shared/components/pie-chart/pie-chart.component.sass [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+
+
+
+
+
+
+
+
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/pie-chart/pie-chart.component.sass:57]: delete empty lines

File: frontend/src/app/shared/components/pie-chart/pie-chart.component.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                name: 'Rest',
+                value: this.totalNumber - this.currentNumber,
+            },
+        ];
+
+        this.colorScheme = {
+            name: 'myScheme',
+            selectable: true,
+            group: ScaleType.Ordinal,
+            domain: [this.activeColor, '#F0F0F0'],
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/pie-chart/pie-chart.component.ts:40]: to a separate (shared) file with colors

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/user/solved-problem/solved-problem.component.sass [hunk 8]
```diff
@@ -1,5 +1,6 @@
 @import 'src/styles/color-variables'
-  
+@import 'src/app/shared/styles/progressbar-colors'
+
```
Comment (tatianahutii) [inline frontend/src/app/modules/user/solved-problem/solved-problem.component.sass:3]: you can move all color variables to one file "color-variables"

File: frontend/src/app/modules/user/solved-problem/solved-problem.component.ts [hunk 9]
```diff
@@ -6,6 +6,8 @@ import { Component, Input, OnInit } from '@angular/core';
     styleUrls: ['./solved-problem.component.sass'],
 })
 export class SolvedProblemComponent implements OnInit {
+    @Input() pieChartActiveColor: string = '#FFA116';
```
Comment (tatianahutii) [inline frontend/src/app/modules/user/solved-problem/solved-problem.component.ts:9]: create a separate fi;e for all color constants that we use across components

Comment (tatianahutii) [reply frontend/src/app/modules/user/solved-problem/solved-problem.component.ts:9]: and use them everywhere

File: frontend/src/app/shared/styles/progressbar-colors.sass [hunk 10]
```diff
@@ -0,0 +1,3 @@
+$emerald-green:#2CBB5D40
+$amber-orange:#FFC01E40
+$crimson-red:#EF474340
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/progressbar-colors.sass:3]: this will cause warnings. Add white space between name and value

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

File: frontend/src/app/modules/user/solved-problem/solved-problem.component.sass [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    background-color: $amber-orange
+    width: 100%    
+
+::ng-deep .level-card:nth-child(3) .progress-bar
+    background-color: $crimson-red
+    width: 100%    
+
+::ng-deep .progress 
+    background-color: transparent
+
```
Comment (apofis96) [inline frontend/src/app/modules/user/solved-problem/solved-problem.component.sass:67]: dont use ::ng-deep without :host

**Review by apofis96 — CHANGES_REQUESTED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**